tests/dtls-client.c

```diff
@@ -216,13 +216,26 @@ send_to_peer(struct dtls_context_t *ctx,
 }
 
 static const dtls_cipher_t* ciphers = NULL;
+static unsigned int force_extended_master_secret = 0;
+static unsigned int force_renegotiation_info = 0;
 
 static void
-get_cipher_suites(struct dtls_context_t *ctx,
-    session_t *session, const dtls_cipher_t **cipher_suites) {
+get_user_parameters(struct dtls_context_t *ctx,
+    session_t *session, dtls_user_parameters_t *user_parameters) {
   (void) ctx;
   (void) session;
-  *cipher_suites = ciphers;
+  user_parameters->force_extended_master_secret = force_extended_master_secret;
+  user_parameters->force_renegotiation_info = force_renegotiation_info;
+  if (ciphers) {
+    int index = 0;
+    while (index <= DTLS_MAX_CIPHER_SUITES) {
```

Comment:
I think that this loop only does the correct thing if ciphers does not contain more than DTLS_MAX_CIPHER_SUITES elements because otherwise, the terminating TLS_NULL_WITH_NULL_NULL will be copied.  Is this guaranteed?

Reply:
The condition protects the destination array `dtls_cipher_t cipher_suites[DTLS_MAX_CIPHER_SUITES + 1];` from overflow.

But I guess, it must be corrected to `index < DTLS_MAX_CIPHER_SUITES` to ensure, that the last entry is the already stored TLS_NULL_WITH_NULL_NULL.